cudax/include/cuda/experimental/__cuco/detail/hash_functions/xxhash.cuh

```diff
@@ -0,0 +1,322 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDAX__CUCO_DETAIL_HASH_FUNCTIONS_XXHASH_CUH
+#define _CUDAX__CUCO_DETAIL_HASH_FUNCTIONS_XXHASH_CUH
+
+#include <cuda/__cccl_config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__bit/bit_cast.h>
+#include <cuda/std/__bit/rotate.h>
+#include <cuda/std/cstddef>
+#include <cuda/std/cstdint>
+#include <cuda/std/span>
+
+#include <cuda/experimental/__cuco/detail/hash_functions/utils.cuh>
+
+#include <cuda/std/__cccl/prologue.h>
+
+namespace cuda::experimental::cuco::__detail
+{
+
+/**
+ * @brief A `_XXHash_32` hash function to hash the given argument on host and device.
+ *
+ * _XXHash_32 implementation from
+ * https://github.com/Cyan4973/xxHash
+ * -----------------------------------------------------------------------------
+ * xxHash - Extremely Fast Hash algorithm
+ * Header File
+ * Copyright (C) 2012-2021 Yann Collet
+ *
+ * BSD 2-Clause License (https://www.opensource.org/licenses/bsd-license.php)
```

Comment:
@jrhemstad did we consider the ramifications of putting BSD-2 licensed code into cccl?